spring.core/src/main/java/com/epam/springcore/task/controller/impl/TraineeController.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.springcore.task.controller.impl;
+
+import com.epam.springcore.task.controller.ITraineeController;
+import com.epam.springcore.task.dto.PassUsernameDTO;
+import com.epam.springcore.task.dto.TraineeDTO;
+import com.epam.springcore.task.dto.TrainerDTO;
+import com.epam.springcore.task.dto.TrainingDTO;
+import com.epam.springcore.task.facade.GymFacade;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/trainees")
+public class TraineeController implements ITraineeController {
+
+    private final GymFacade gymFacade;
+
+    @PostMapping("/register")
+    @Override
+    public ResponseEntity<Object> registerTrainee(@RequestBody TraineeDTO traineeDTO){
```

Comment:
Don't use ResponseEntity, by default Spring returns HTTP 200 for successful cases, in other cases use @ResponseStatus annotation, like this method, when you need return HTTP 201